### ISSUE TITLE
ENH: improve 'explode' to work on GeoDataFrames (preserving other columns)

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#667]
Currently the `explode` method actually works on both GeoSeries and GeoDataFrame, but when using on a GeoDataFrame, it actually performs it only on the geometry column and all other columns are lost. 
It would be a nice enhancement to preserve the other columns (and repeating the values for those rows that have been exploded).

Small example:

```
In [1]: import geopandas
   ...: from shapely.geometry import MultiPoint, Point

In [2]: s = geopandas.GeoSeries([MultiPoint([Point(1, 2), Point(2, 3)]), Point(5, 5)])

In [3]: s.explode()
Out[3]: 
0  0    POINT (1 2)
   1    POINT (2 3)
1  0    POINT (5 5)
dtype: object

In [4]: df = geopandas.GeoDataFrame({'geometry': s, 'col': [1, 2]})

In [5]: df
Out[5]: 
   col                    geometry
0    1  (POINT (1 2), POINT (2 3))
1    2                 POINT (5 5)

In [6]: df.explode()
Out[6]: 
0  0    POINT (1 2)
   1    POINT (2 3)
1  0    POINT (5 5)
dtype: object

```

To manually merge original columns with exploded geoms, you can do something like this:

```
In [7]: geoms_exploded = df.explode().reset_index(level=1, drop=True)

In [8]: df.drop(columns='geometry').join(geoms_exploded.rename('geometry'))
Out[8]: 
   col     geometry
0    1  POINT (1 2)
0    1  POINT (2 3)
1    2  POINT (5 5)
```

but IMO this should be the default behaviour of `GeoDataFrame.explode(..)`

**Comment by cr458** [comment]
Could have a look at this if needed?

**Comment by jorisvandenbossche** [comment]
Needed not, but welcome to do a PR certainly!

**Comment by jorisvandenbossche** [comment]
Closed by #671
